### PR TITLE
Fix Horace pinning issue, emotes in accountability station, focused raider

### LINKED ISSUE #26: Emoji reactions in accountability station
**Is your feature request related to a problem? Please describe.**
When people write a message in #accountability-station, he no longer reacts with a random encouraging emoji, I'd love for this feature to return!

**Describe the solution you'd like**
I want Horace to react with emojis to all tasks containing a sort of checkmark in the accountability channel, to help encourage people to finish more tasks.

**Describe alternatives you've considered**
No alternatives.

**Additional context**

### LINKED ISSUE #40: Focused Raider Feature
The Focused Raider role used to block all the channels except accountability station i think and choose your roles. I might be missing some information here but ithe feature is needed :D 

**Is your feature request related to a problem? Please describe.**
People get distracted when they open discord to raid so focused raider helps to block out all the channels except accountability station and choose your roles (i think it was just these two channels  I'll ask and let you know) so it minimises distraction.

**Describe the solution you'd like**
The ability to react to the :letsgo: emoji in the choose your roles channel so that one can enter focused raider and work peacefully.

**Describe alternatives you've considered**
I haven't thought of any alternatives to this yet. Sorry!

**Additional context**
Go ahead and provide us with additional context, or even screenshots, of what this might look like! For example, you type a command and a friend pretends to be the bot! :)

### LINKED ISSUE #72: Horace sends a double confirmation when pinning pinning
**Please Describe Your Issue**
When something is pinned in accountability station, Horace will post a message - plus discord will inform about it.
Only one is needed.

**How to Reproduce**
1. Pin something
2. See two messages with same information

**Expected behavior**
Only discord should be shown

**Extra Context**

image

### PR DESCRIPTION
**Pull request by Dendrobyte**
## Resolves #72 
- Horace now unpins when you re-react with the pushpin emote (he leaves a pin for you to click)
- Only Discord message shows when pinning a message

## Resolves #40 
- React with the :LETSGO: reaction to a specific message in a choose-your-roles channel to get the focused raider, and remove the reaction to get rid of it.

## Resolves #26 
- Horace now adds reactions in accountability station from a set of emojis, with language support

### CODE REVIEW

**Comment by Dendrobyte** [comment]
@Moesgaarda You'll need to put both choose-your-roles and the message id of the message to react to in the configuration. It pulls the actual reaction from `hocReaction`

File: eventActions/accountabilityActions.js [hunk 1]
```diff
@@ -8,37 +20,50 @@ class accountabilityActions {
...
+				if(!isMessagePinnedAtAll(sentMessage, fetchedPins)){
+					// Pin the message
+					var existingMessageCount = 0;
 
-				for (var i = 0; i < fetchedPins.size; i++){
-					var msgVal = pinMsgIterator.next().value;
-					if(msgVal.author.id === user.id){
-						existingMessageCount++;
+					// Let the Discord message inform a user something was pinned
+					// currentChannel.send('Hey, ' + user.username + ', I\'ve pinned your message for you as requested!');
```
Comment (Moesgaarda) [inline eventActions/accountabilityActions.js]: Remove uncommented code

File: eventActions/accountabilityActions.js [hunk 2]
```diff
@@ -74,17 +98,43 @@ class accountabilityActions {
...
+		// Flag emotes
+		var length = random_emotes.length;
+		var flags = [
+			{ language: 'french', emote: '🇫🇷'},
+			{ language: 'spanish', emote: '🇪🇸'},
+			{ language: 'italian', emote: '🇮🇹'}
+		];
+
+		// Define special emotes (I didn't want to put all of them in the configuration...)
+		var checkmark = '562072692086276118';
```
Comment (Moesgaarda) [inline eventActions/accountabilityActions.js]: I think this is already in config, they should generally be there, so it's easy to change them in the future.

File: eventActions/accountabilityActions.js [hunk 3]
```diff
@@ -8,37 +20,50 @@ class accountabilityActions {
...
-			});
+					// Pin the message
+					sentMessage.clearReactions();
+					sentMessage.pin();
 
-			// Pin the message
-			sentMessage.pin();
+					// If they have other pinned messages, give them a good 'ol reminder.
+					if (existingMessageCount > 1){
+						currentChannel.send('Hey, ' + user.username + ', I just wanted to remind you that you have ' + existingMessageCount + ' other pinned messages :smile:');
```
Comment (Moesgaarda) [inline eventActions/accountabilityActions.js]: Use 😄 smile emoji instead for consistency

File: eventActions/accountabilityActions.js [hunk 4]
```diff
@@ -1,5 +1,17 @@
 const config = require('../config.json');
 
+// Helper method to loop through pins. I was getting mad when I named it...
+function isMessagePinnedAtAll(messageToCheck, setOfPinnedMessages){
+	var fetchedMessagesIterator = setOfPinnedMessages.values();
+	var msgVal = fetchedMessagesIterator.next().value;
```
Comment (Moesgaarda) [inline eventActions/accountabilityActions.js]: Use const or let instead of var

Comment (Moesgaarda) [reply eventActions/accountabilityActions.js]: (in general, this should be fixed)

Comment (Dendrobyte) [reply eventActions/accountabilityActions.js]: the `msgVal` field changes, but iterator is now `const` 👌

File: eventActions/accountabilityActions.js (hunk 2 above)
Comment (Dendrobyte) [reply eventActions/accountabilityActions.js]: The checkmark defined in configuration is the normal ✅ but I'll add "pom" and "yes2" as options in the config for these ones.

File: eventActions/accountabilityActions.js (hunk 1 above)
Comment (Dendrobyte) [reply eventActions/accountabilityActions.js]: The more I look at this the more I think it means the code that was commented out

Done!

File: eventActions/accountabilityActions.js (hunk 2 above)
Comment (Dendrobyte) [reply eventActions/accountabilityActions.js]: Variables stored in configuration, pom and yes2 added to config

File: eventActions/accountabilityActions.js [hunk 5]
```diff
@@ -74,17 +96,43 @@ class accountabilityActions {
...
-				if(curMessage.pinned){
-					curMessage.unpin();
-					curMessage.channel.send('Hey, ' + user.username + ', I\'ve unpinned your message as requested!');
-				}
+	// Add a random reaction to a message sent
+	static async addReaction(client, message){
+		if(message.channel.id != config.channels.accountability) return;
+		if(message.content.toLowerCase().includes('!unpin')) return;
+		// Define an array of emojis to pull from
+		var random_emotes = config.emotes.accountability_emotes_array;
```
Comment (Moesgaarda) [inline eventActions/accountabilityActions.js]: Vars and I think you forgot to add the new array to the config file :D

Comment (Dendrobyte) [reply eventActions/accountabilityActions.js]: Ah right- forgot to add it to example ;-;

**Pull request merged**